app/src/pages/CounterPage/containers/CounterPageContainer.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from "react";
+import Counter from "../components/Counter";
+
+class CounterPageContainer extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      countValue: 0,
+      parityType: "even",
+    };
+
+    console.log("CONSTRUCTOR");
+  }
+
+  componentDidMount() {
+    console.log("DID MOUNT");
+  }
+
+  componentDidUpdate() {
+    console.log(this.state.parityType);
+  }
+  shouldComponentUpdate() {
+    console.log("SHOULD COMPONENT UPDATE");
+    return true;
+  }
+
+  componentWillUnmount() {
+    console.log("DESTROYED");
+  }
+
+  handleIncrement = () => {
+    this.setState((state) => {
+      const incrementedValue = state.countValue + 1;
+      const isEven = incrementedValue % 2 === 0 ? "even" : "odd";
+      return {
+        countValue: incrementedValue,
+        parityType: isEven,
+      };
+    });
+  };
+
+  handleDecrement = () => {
+    this.setState((state) => {
+      const incrementedValue = state.countValue - 1;
+      const isEven = incrementedValue % 2 === 0 ? "even" : "odd";
```

Comment:
rename this variable. Your variable's name means bool value but here is a string